cmd/cmd_bucket.go

```diff
@@ -271,8 +271,16 @@ func listBuckets(ctx *cli.Context) error {
 	c, cancelCreateBucket := context.WithCancel(globalContext)
 	defer cancelCreateBucket()
 
-	bucketListRes, err := client.ListBuckets(c, sdktypes.EndPointOptions{})
+	spInfo, err := client.ListStorageProviders(c, true)
+	if err != nil {
+		fmt.Println("fail to get SP info to list bucket:", err.Error())
+		return nil
+	}
 
+	bucketListRes, err := client.ListBuckets(c, sdktypes.ListBucketsOptions{false, &sdktypes.EndPointOptions{
+		Endpoint:  spInfo[0].Endpoint,
```

Comment:
what is the `Endpoint` used for?

Reply:
The go-sdk interface has some issues, including a null pointer error. Therefore, it is necessary to initialize and pass an endpoint pointer to specify which service provider's meta service to access.